### PR TITLE
Add CI

### PR DESCRIPTION
**Pull request by paladox**
* Bumps MW requirements to 1.39+.
* Requires PHP 7.4+
* Adds CI to run tests
* Fixes tests to support MW 1.39
* Adapts to changes done to the naming of files.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
+
+jobs:
+
+  test:
+
+    runs-on: ubuntu-22.04
```
Comment (JeroenDeDauw) [inline .github/workflows/ci.yml:14]: I know you copied, but I started wondering why we are not using 24.04

Comment (paladox) [reply .github/workflows/ci.yml:14]: Oh! I can do that. Thanks!

Comment (JeroenDeDauw) [reply .github/workflows/ci.yml:14]: Or latest for that matter. That's what I do for all of my extensions, see example https://github.com/ProfessionalWiki/PageApprovals/blob/master/.github/workflows/ci.yml#L23

Comment (JeroenDeDauw) [reply .github/workflows/ci.yml:14]: Been doing that for years and never had a problem

**Comment by JeroenDeDauw** [comment]
Looks good so far, though it gets a bit more difficult to review due to the PR containing a bunch of unrelated changes. I'm mentioning it for future PRs and am not requesting you change this one.

**Comment by paladox** [comment]
Coverage and MW 1.40+ ci will be added separately. For now the tests are now working!!

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+export
+
+# setup for docker-compose-ci build directory
+# delete "build" directory to update docker-compose-ci
+
+ifeq (,$(wildcard ./build/))
+    $(shell git submodule update --init --remote)
+endif
+
+EXTENSION=SemanticResultFormats
```
Comment (JeroenDeDauw) [inline Makefile:11]: oops ;)

Comment (paladox) [reply Makefile]: Oh, thanks!

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+                   
+      - name: Update submodules
+        run: git submodule update --init --remote
+
+      - name: Run tests
+        run: make ci
+        if: matrix.coverage == false
```
Comment (JeroenDeDauw) [inline .github/workflows/ci.yml:47]: Not specific to this PR, but I noticed that the "Run tests" task output if 90%+ setup noise. So it if you open up the output on GitHub actions, you have to do a lot of scrolling. Ideally setup is done in a dedicated step.

@gesinn-it-gea

Comment (JeroenDeDauw) [reply .github/workflows/ci.yml:47]: See https://github.com/SemanticMediaWiki/SemanticWatchlist/actions/runs/12561101773/job/35019462246

**Pull request merged**